runtime/src/offchaincb.rs

```diff
@@ -186,4 +186,219 @@ impl<T: Trait> Module<T> {
 			}
 		})
 	}
-}
\ No newline at end of file
+}
+
+#[cfg(test)]
+mod tests {
+	use codec::Decode;
+	use std::sync::Arc;
+	use parking_lot::RwLock;
+	use sr_primitives::{
+		Perbill, generic, RuntimeAppPublic,
+		testing::{Header, TestXt, UintAuthorityId},
+		traits::{IdentityLookup, BlakeTwo256, Block, Dispatchable},
+	};
+	use offchain::testing::TestOffchainExt;
+	use primitives::{H256, Blake2Hasher};
+	use support::{construct_runtime, parameter_types, assert_ok};
+	use runtime_io::{with_externalities, TestExternalities};
+	use system;
+	use crate::offchaincb as offchaincb;
+
+	type AccountId = u64;
+	type AccountIndex = u64;
+	type Extrinsic = TestXt<Call, ()>;
+	type SubmitTransaction = system::offchain::TransactionSubmitter<UintAuthorityId, Call, Extrinsic>;
+
+	// TODO: implement this for runtime or call?
```

Comment:
Yes I have to look into this again + I am _totally_ mocking the existence of a signature (basically `signature = (address, ())`). Is this too far even for tests? 😁